### PR TITLE
feat(core): Add 'Get-Encoding()' function to fix missing webClient encoding

### LINKED ISSUE #4911: [Bug] Can not install with url encoded in `utf-8`
#### Current Behavior

图片

#### Expected Behavior

The file is encoding with utf-8, but the `Webclient` use `default` encoding, and it is not recommended.

See more in [system.net.webclient.encoding](https://docs.microsoft.com/en-us/dotnet/api/system.net.webclient.encoding?view=net-6.0#system-net-webclient-encoding) and [system.text.encoding.default](https://docs.microsoft.com/en-us/dotnet/api/system.text.encoding.default?view=net-6.0#system-text-encoding-default).

#### Possible Solution

We can solve this problem by set the `Encoding` for `Webclient`.

### System details

**Windows version:** [e.g. 7, 8, 10]

win10 

**OS architecture:** [e.g. 32bit, 64bit]

64bit

**PowerShell version:** [output of `"$($PSVersionTable.PSVersion)"`]

5.1.19041.1645

**Additional software:** [(optional) e.g. ConEmu, Git]

#### Scoop Configuration

default setting.

### PR DESCRIPTION
**Pull request by yi-Xu-0100**
Add Get-Encoding function in core.ps1

#### Motivation and Context

Closes #4911 #4324

Relates to #4911 #4324

#### Checklist:

- [x] I have read the [Contributing Guide](https://github.com/ScoopInstaller/.github/blob/main/.github/CONTRIBUTING.md).
- [x] I have ensured that I am targeting the `develop` branch.
- [x] I have updated the documentation accordingly.
- [x] I have updated the tests accordingly.
- [x] I have added an entry in the CHANGELOG.

### CODE REVIEW

**Comment by yi-Xu-0100** [comment]
@rashil2000 Any review here will be helpful~ 😁

File: lib/core.ps1 [hunk 1]
```diff
@@ -15,6 +15,15 @@ function Optimize-SecurityProtocol {
     }
 }
 
+function Get-Encoding($url) {
```
Comment (niheaven) [inline lib/core.ps1:18]: Pass `$wc` in the function

File: lib/manifest.ps1 [hunk 2]
```diff
@@ -1,3 +1,4 @@
+. "$PSScriptRoot\..\lib\core.ps1"
 function manifest_path($app, $bucket) {
```
Comment (niheaven) [inline lib/manifest.ps1:2]:
```suggestion
function manifest_path($app, $bucket) {
```

File: libexec/scoop-virustotal.ps1 [hunk 3]
```diff
@@ -34,6 +34,7 @@
 #   -n, --no-depends          By default, all dependencies are checked too. This flag avoids it.
 #   -u, --no-update-scoop     Don't update Scoop before checking if it's outdated
 
+. "$PSScriptRoot\..\lib\core.ps1"
 . "$PSScriptRoot\..\lib\getopt.ps1"
```
Comment (niheaven) [inline libexec/scoop-virustotal.ps1:38]:
```suggestion
. "$PSScriptRoot\..\lib\getopt.ps1"
```

**Review by niheaven — CHANGES_REQUESTED** [review]

**Comment by niheaven** [comment]
Do you have some testing manifests?

**Comment by yi-Xu-0100** [comment]
@niheaven ,Thanks for your reply. 😁

In https://github.com/ScoopInstaller/Scoop/issues/4911#issue-1229175577 , I use 

```powershell
scoop install "https://raw.githubusercontent.com/ygguorun/scoop-bucket/master/bucket/PDFPatcher.json"
```

, and [this manifest - DoveBoy/Apps/pdfpatcher](https://raw.githubusercontent.com/DoveBoy/Apps/master/bucket/pdfpatcher.json) has the `checkver`, and they all work well.

**Comment by niheaven** [comment]
`checkver` has some error, could you please check it again?

**Comment by yi-Xu-0100** [comment]
@niheaven  I used it for `scoop update` and `scoop install`, they all work well. 

Could you please provider the way to test the `checkver`, I have no idea for it.🤔

**Comment by niheaven** [comment]
Checkout this branch and cd in any bucket, run `.\bin\checkver.ps1 xxx`, there are some bugs :)

**Comment by yi-Xu-0100** [comment]
@niheaven Thanks for pointing out the problem, I think it's fixed now. 😁

**Review by niheaven — APPROVED** [review]

**Comment by yi-Xu-0100** [comment]
@niheaven Thanks for adding the judgment for null values, but there is something we need to revert.

If we use `(Get-Encoding($wc)).GetString($wc.DownloadData($url))`, then `(Get-Encoding($wc))` can only get the return result in UTF-8, because The `ResponseHeaders` of `$wc` will be `$null` before the data is downloaded. 

So I re-split the expression to handle the case where `charset` is something else.

**Comment by niheaven** [comment]
Aha, right?

Okay, I'll test it tomorrow and it seems stable enough now.

**Review by niheaven — APPROVED** [review]

**Pull request merged**